snap/validate.go

```diff
@@ -117,12 +117,8 @@ func ValidateApp(app *AppInfo) error {
 	}
 
 	// Validate app name
-	if app.Name == "" {
-		return fmt.Errorf("snap app name cannot be empty")
-	}
-
 	if !validAppName.MatchString(app.Name) {
-		return fmt.Errorf("invalid snap app name: %q", app.Name)
+		return fmt.Errorf("cannot use %q as app name (app names are letters/digits with dash as separator)", app.Name)
```

Comment:
Sorry for the nitpick, but can we please avoid the cacophony here with something like:

"cannot have %q as app name - use letters, digits, and dash as separator"